orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/AbstractInstancesCheckTask.groovy

```diff
@@ -119,7 +126,16 @@ abstract class AbstractInstancesCheckTask extends AbstractCloudProviderAwareTask
         }
       }
 
-      verifyServerGroupsExist(stage)
+      try {
+        verifyServerGroupsExist(stage)
+      } catch (MissingServerGroupException e) {
+        if (waitForUpServerGroup()) {
```

Comment:
I figured out they maintain state during the execution of the task, and added a 10 minute timeout.